src/lib/router.js

```diff
@@ -33,6 +35,10 @@ async function swapContent(url) {
     return;
   }
 
+  // Load the entrypoint simultaneously with content.
+  const entrypointPromise = entrypointForRoute(url);
```

Comment:
Maybe we should split `swapContent` so it contains two functions, one that handles fetching the js and one that handles fetching the html. Doing both things in the same function makes it a little confusing.

Reply:
The main thing I want to get across here is that the loads can happen in parallel. I can refactor it so it's two separate functions if you like, but we end up just awaiting on both results regardless.

Reply:
yep, understood. I was thinking it might be good as two functions in case we continue to add logic to it. Being able to separate at a glance which parts are related to loading the js and which parts are related to loading the html might be helpful.